src/dbnode/x/xio/types.go

```diff
@@ -87,9 +87,9 @@ type ReaderSliceOfSlicesIterator interface {
 	// Size gives the size of bytes in this iterator.
 	Size() (int, error)
 
-	// Rewind returns the iterator to the beginning.
-	// This operation is invalid if any of the block readers have been read.
-	Rewind()
+	// Returns a new copy of the iterator with same internals
+	// so it can be iterated without progressing the original.
```

Comment:
where is the clone pointing at? at the beginning or where this iterator is currently pointing?

Reply:
Yeah we want to point to specifically where the iterator was when it was passed to us. I'll update comment to make clear.

But yeah the root problem here was e.g. an iterator comes in at idx of 1, but then we rewind to 0 and so we've changed that state for whoever uses it next from the state it was intended. What I'd like to understand is why it is valid for us to receive an iterator already at idx of 1.

Reply:
Hm, given that we know what's causing it here (makes sense with that weird iterator that starts at 1 sometimes to not want to be rewinded to 0), can we stick with an updated `Rewind` semantic to avoid allocs? 

Instead we can refactor it to `ResetToIndex(idx int)`, and also expose a `CurrentIndex() int` method here; then can avoid thinking about life cycles for the cloned iterators, or accidentally allocing, etc.

Reply:
Aren't these allocs cheap? They're just struct wrappers per series right? Obv it is non-zero cost, but the rewinding is also pretty a bit confusing and ideally we wouldn't be mutating any state at all to be safest.